[#28] RestaurantService 테스트 작성

seohoon-dev opened the pull request.

seohoon-dev committed:
[#28] RestaurantService 테스트 작성

seohoon-dev committed:
[#30] OwnerService 테스트 작성

seohoon-dev committed:
[#32] MenuService 테스트 작성

f-lab-dev commented on src/test/java/com/hoon/foodrocket/service/RestaurantServiceTest.java:
```diff
@@ -0,0 +1,219 @@
+        restaurantService.getRestaurantsByAddressAndCategory(region, category, cursorId);
+
+        // then
+        fail("예외가 발생해야 한다.");
```
이런것도 테스트를 작성하면서 발견하게되는 것중에 하나인데요.
리스트를 조회했을때 검색결과가 없으면 빈 리스트가 리턴되는게 좋지 않을까요?

seohoon-dev replied:
음.. 아래처럼 조회 결과를 받는걸 말씀하시는거 맞나요?
`List<Restaurant> list = restaurantService.getRestaurantsByAddressAndCategory(...);`

이럴 경우에 예외가 정상적으로 발생하면 then 부분까지 안가고 테스트가 종료되는데  그러면 조회 결과인 list를 사용하지 않고 끝나게 되서 불필요하게 변수를 할당한다고 생각됩니다.

현재 이런식으로 생각이 되는데 빈 리스트를 반환하는 이유가 뭔지 알려주시면 감사하겠습니다. :)

seohoon-dev committed:
Merge pull request #33 from f-lab-edu/issue/32

[#32] MenuService 테스트 작성

seohoon-dev committed:
Merge pull request #31 from f-lab-edu/issue/30

[#30] OwnerService 테스트 작성

seohoon-dev committed:
리뷰 내용 반영

- 검색 결과가 없으면 빈 목록을 반환한다.
- '가게목록_실패_가게정보_없음' 테스트 코드는 빈 목록을 반환하는 것이 실패 유형이 아니기 때문에 제거한다.

f-lab-dev commented on src/test/java/com/hoon/foodrocket/service/RestaurantServiceTest.java:
```diff
@@ -55,24 +55,6 @@
     }
 
-    @Test(expected = IllegalStateException.class)
-    public void 가게목록_실패_가게정보_없음() throws Exception {
```
실패 케이스라기보다는 "검색결과가 없을 시 빈 리스트를 리턴한다"의 케이스로 테스트를 작성해보시는건 어떨까요?

테스트는 잘 돌아가는 케이스말고도 여러가지 상황의 입력과 결과를 정의해줘야합니다~

seohoon-dev committed:
리뷰 내용 반영

- 가게 목록을 호출했을 때 검색 결과가 없다면 빈 목록을 반환한다.
